### PR TITLE
CI(suggestions): Extract uploading of code suggestions with a local composite action

### PR DESCRIPTION
**Pull request by echoix**
Based on https://github.com/OSGeo/grass-addons/pull/1045

This PR extracts the logic of uploading suggestions into a composite action that can be used multiple times.
Afterwards, the grass-addons repo could use the action defined from here instead of locally, and would stay updated.

Next PR will use this to add posting suggestions from black, without duplicating any code.

### CODE REVIEW

**Comment by echoix** [comment]
My next PR is ready, but I'm waiting for this to be merged, so I can rebase it on top of it, since it is required to have the action available.

**Comment by echoix** [comment]
This is my second review priority. If you want to see it in action, you can see it in action in the grass-addons

**Review by nilason — APPROVED** [review]

**Pull request merged**
